web/impact/impact/tests/test_accelerator_routes.py

```diff
@@ -0,0 +1,174 @@
+# MIT License
+# Copyright (c) 2017 MassChallenge, Inc.
+
+from test_plus.test import TestCase
+from impact.urls import accelerator_router
+from django.apps import apps
+from rest_framework.test import APIClient
+from django.contrib.contenttypes.models import ContentType
+from impact.tests.factories.accelerator import (
+    StartupFactory,
+    IndustryFactory,
+    OrganizationFactory)
+from django.contrib.auth.models import Permission
+import simplejson as json
+
+from impact.tests.factories import (
+    UserFactory,
+    StartupStatusFactory
+)
+from impact.tests.factories import ContentTypeFactory
+from impact.tests.factories import PermissionFactory
+from accelerator.models import Startup
+
+
+class TestAcceleratorRoutes(TestCase):
+    client_class = APIClient
+    user_factory = UserFactory
+
+    @classmethod
+    def setUpClass(cls):
+        ContentTypeFactory(app_label='mc', model='startup')
+        ContentTypeFactory(app_label='mc', model='startupstatus')
+        ContentTypeFactory(app_label='mc', model='organization')
+        ContentTypeFactory(app_label='mc', model='recommendationtag')
+        ContentTypeFactory(app_label='mc', model='programrole')
+
+    @classmethod
+    def tearDownClass(cls):
+        ContentType.objects.filter(
+            app_label='mc',
+            model__in=[
+                'startup',
+                'organization',
+                'programrole',
+                'recommendationtag']).delete()
+
+    def test_user_with_permissions_can_create_objects(self):
+        url_name = "object-list"
+        StartupStatusFactory(id=1)
+        view_kwargs = {
+            'app': 'accelerator',
+            "model": "startup",
+        }
+        self.response_401(self.get(url_name, **view_kwargs))
+        startup_add_permission, _ = Permission.objects.get_or_create(
+            content_type=ContentType.objects.get(
+                app_label='mc',
+                model='startup'),
+            codename='add_startup',
+        )
+        perm_user = self.make_user(
+            'perm_user2@test.com', perms=["mc.add_startup"])
+        perm = PermissionFactory.create(codename='change_startup')
+        view_perm = PermissionFactory.create(codename='view_startup')
+        startup_permission, _ = Permission.objects.get_or_create(
+            content_type=ContentType.objects.get(
+                app_label='mc',
+                model='startup'),
+            codename='view_startup',
+        )
+        perm_user.user_permissions.add(perm)
+        perm_user.user_permissions.add(startup_add_permission)
+        perm_user.user_permissions.add(startup_permission)
+        perm_user.user_permissions.add(view_perm)
+        perm_user.save()
```

Comment:
I find it very weird that Permission defined on `mc.startup` controls the permissions for `accelerator.startup`. This is a major blocker in my view.

Reply:
this will unfortunately need to be the case for now due to how internally we reference the models and the fact that they are unmanaged. the 'mc.startup' is in reference to the table name from which the models are derived from. I'm not sure how to get around this limitation and be able to use unmanaged models atop existing data.

Reply:
I understand the problem. Then what I'd do is name somewhere `settings.py`-y `ACC_APP_NAME = mc` and use that wherever. Otherwise it's just too confusing.